src/games/supported/DemonAttack.cpp

```diff
@@ -121,3 +124,37 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect DemonAttackSettings::getAvailableModes() {
+    ModeVect modes = {1, 3, 5, 7};
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void DemonAttackSettings::setMode(game_mode_t m, System &system,
+                              std::unique_ptr<StellaEnvironmentWrapper> environment) {
+    if(m == 1 || m == 3 || m == 5 || m == 7) {
+        // read the mode we are currently in
+        unsigned char mode = readRam(&system, 0xEA);
+        // press select until the correct mode is reached
+        while (mode != m) {
+            m_level_change = true;
```

Comment:
This seems unnecessary. Doesn't it get set below?

Reply:
Done.